src/main/modules/ardoq/NPMParser.ts

```diff
@@ -3,24 +3,65 @@ import { HTTPError } from '../../HttpError';
 import { Dependency } from './Dependency';
 import { IParser } from './IParser';
 
+import {
+  NodeLockfileVersion,
+  getNpmLockfileVersion,
+  parseNpmLockV2Project,
+  parsePkgJson,
+} from 'snyk-nodejs-lockfile-parser';
+
 export class NPMParser implements IParser {
-  public extractTopTierDeps(depString: string): Dependency[] {
+  public async extractTopTierDeps(packageLock: string, packageJson?: string | undefined): Promise<Dependency[]> {
+    const dependencies: Dependency[] = [];
     try {
-      const json = JSON.parse(depString);
+      const lockVersion = getNpmLockfileVersion(packageLock);
+
+      if (lockVersion == NodeLockfileVersion.NpmLockV1) {
```

Comment:
It may be better to separate the NPM parser into 2 separate implementations? they don't seem to share much logic.

Reply:
Yeah, I thought this too - it might actually be better to `throw` if we encounter a v1 `package-lock.json` file because the snyk-parser won't parse it. And it's likely to blow up with 1000s of deps if it does encounter a v1 file